config.ru

```diff
@@ -1,4 +1,20 @@
 # This file is used by Rack-based servers to start the application.
 
+if ENV['RAILS_ENV'] == 'production'
+  require 'unicorn/worker_killer'
+
+  max_request_min =  500
+  max_request_max =  600
+
+  # Max requests per worker
+  use Unicorn::WorkerKiller::MaxRequests, max_request_min, max_request_max
+
+  oom_min = (240) * (1024**2)
+  oom_max = (260) * (1024**2)
+
+  # Max memory size (RSS) per worker
+  use Unicorn::WorkerKiller::Oom, oom_min, oom_max
```

Comment:
Well, they are supposed to run forever.    But when memory leaks happen (which from my shallow research seems most often than not with rails), these works keep consuming more and more.   This code periodically kills workers (which the master processes restrarts) to free up memory.
https://www.digitalocean.com/community/tutorials/how-to-optimize-unicorn-workers-in-a-ruby-on-rails-app